packages/pubsub/__tests__/AWSAppSyncRealTimeProvider.test.ts

```diff
@@ -0,0 +1,879 @@
+					expect.assertions(1);
+
+					const observer = provider.subscribe('test', {
+						appSyncGraphqlEndpoint: 'ws://localhost:8080',
+					});
+
+					const subscription = observer.subscribe({
+						error: x => {},
+					});
+
+					await fakeWebSocketInterface?.readyForUse;
+					await fakeWebSocketInterface?.triggerOpen();
+					await fakeWebSocketInterface?.sendDataMessage({
+						type: MESSAGE_TYPES.GQL_CONNECTION_ERROR,
+						payload: {
+							errors: [
+								{
+									errorType: 'Retriable Test',
+									errorCode: 408, // Request timed out - retriable
+								},
+							],
+						},
+					});
+
+					// Wait while the message is received and processed
+					await fakeWebSocketInterface?.wait(50);
+
+					// Watching for raised exception to be caught and logged
+					expect(loggerSpy).toBeCalledWith(
+						'DEBUG',
+						'error on bound ',
+						expect.objectContaining({
+							message: expect.stringMatching('Retriable Test'),
+						})
+					);
+				});
+
+				test('subscription observer error is triggered when a connection is formed and an ack data message is received then ack timeout prompts disconnect', async () => {
+					expect.assertions(1);
+
+					const observer = provider.subscribe('test', {
+						appSyncGraphqlEndpoint: 'ws://localhost:8080',
+					});
+
+					const subscription = observer.subscribe({
+						error: () => {},
+					});
+
+					await fakeWebSocketInterface?.readyForUse;
+					await fakeWebSocketInterface?.triggerOpen();
+					await fakeWebSocketInterface?.sendMessage(
+						new MessageEvent('connection_ack', {
+							data: JSON.stringify({
+								type: MESSAGE_TYPES.GQL_CONNECTION_ACK,
+								payload: { connectionTimeoutMs: 20 },
+							}),
+						})
+					);
+					// Wait for the connection timeout to get updated with the keep alive payload
+					await fakeWebSocketInterface?.wait(10);
+
+					await fakeWebSocketInterface?.sendDataMessage({
+						type: MESSAGE_TYPES.GQL_CONNECTION_KEEP_ALIVE,
+						payload: { data: {} },
+					});
+
+					// Now wait for the timeout to elapse
+					await fakeWebSocketInterface?.wait(100);
+
+					expect(loggerSpy).toBeCalledWith(
+						'DEBUG',
+						'Disconnect error: Timeout disconnect'
+					);
+				});
+
+				test('socket is closed when subscription is closed', async () => {
+					expect.assertions(1);
+
+					const observer = provider.subscribe('test', {
+						appSyncGraphqlEndpoint: 'ws://localhost:8080',
+					});
+
+					const subscription = observer.subscribe({});
+
+					await fakeWebSocketInterface?.readyForUse;
+					await fakeWebSocketInterface?.triggerOpen();
+					await fakeWebSocketInterface?.handShakeMessage();
+					await fakeWebSocketInterface?.sendDataMessage({
+						type: MESSAGE_TYPES.GQL_DATA,
+						payload: { data: {} },
+					});
+					await subscription.unsubscribe();
+					await fakeWebSocketInterface?.triggerClose();
+
+					expect(fakeWebSocketInterface.hasClosed).resolves.toBe(undefined);
```

Comment:
Nit: You could use the built-in function
```suggestion
					expect(fakeWebSocketInterface.hasClosed).resolves.toBeUndefined();
```